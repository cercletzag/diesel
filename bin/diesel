#!/bin/sh
cd diesel_cli && cargo run -- $@
