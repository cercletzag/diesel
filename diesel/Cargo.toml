[package]
name = "diesel"
version = "0.4.1"
authors = ["Sean Griffin <sean@seantheprogrammer.com>"]
license = "MIT OR Apache-2.0"
description = "A safe, extensible ORM and Query builder"
documentation = "http://sgrif.github.io/diesel/diesel/index.html"
homepage = "https://github.com/sgrif/diesel"
repository = "https://github.com/sgrif/diesel"
keywords = ["orm", "database", "postgres", "postgresql", "sql"]

[dependencies]
libc = "0.2.*"
pq-sys = { version = "^0.2.0", optional = true }
libsqlite3-sys = { version = "^0.4.0", optional = true }
byteorder = "0.3.*"
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git", optional = true }
chrono = { version = "^0.2.17", optional = true }

[dev-dependencies]
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git" }
dotenv = "^0.6.0"
tempdir = "^0.3.4"

[features]
default = ["postgres"]
unstable = []
large-tables = []
postgres = ["pq-sys"]
sqlite = ["libsqlite3-sys"]
