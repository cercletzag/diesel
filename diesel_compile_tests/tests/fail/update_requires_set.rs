extern crate diesel;

use diesel::*;

table! {
    users {
        id -> Integer,
    }
}

fn main() {
    let conn = SqliteConnection::establish(":memory:").unwrap();
    update(users::table)
        .execute(&conn);
}
