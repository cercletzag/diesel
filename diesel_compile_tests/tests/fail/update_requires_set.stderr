error[E0277]: the trait bound `diesel::query_builder::update_statement::SetNotCalled: diesel::query_builder::QueryFragment<_>` is not satisfied
  --> $DIR/update_requires_set.rs:14:10
   |
14 |         .execute(&conn);
   |          ^^^^^^^ the trait `diesel::query_builder::QueryFragment<_>` is not implemented for `diesel::query_builder::update_statement::SetNotCalled`
   |
   = note: required because of the requirements on the impl of `diesel::query_builder::QueryFragment<_>` for `diesel::query_builder::UpdateStatement<users::table, diesel::query_builder::where_clause::NoWhereClause>`
   = note: required because of the requirements on the impl of `diesel::query_dsl::load_dsl::ExecuteDsl<_, _>` for `diesel::query_builder::UpdateStatement<users::table, diesel::query_builder::where_clause::NoWhereClause>`
